grafeas/MIGRATING.md

```diff
@@ -0,0 +1,364 @@
+each specific client instance. For example, to set default credentials and
+timeout for all Grafeas V1 clients:
+
+```
+Grafeas::V1::Grafeas::Client.configure do |config|
+  config.credentials = my_credentials
+  config.timeout = 10.0
+end
+```
+
+Individual RPCs can also be configured independently. For example, to set the
+timeout for the `create_occurrence` call:
+
+```
+Grafeas::V1::Grafeas::Client.configure do |config|
+  config.rpcs.create_occurrence.timeout = 20.0
+end
+```
+
+Finally, you can override the configuration for each client instance. See the
+section on [Creating Clients](#creating-clients) for details.
+
+### Creating Clients
+
+In older releases, to create a client object, you would use the
+`Grafeas.new` class method. Keyword arguments were available to
+select a service version and to configure parameters such as credentials and
+timeouts.
+
+With the 1.0 release, use the `Grafeas.grafeas` class
+method to create a client object. You may select a service version using the
+`:version` keyword argument. However, other configuration parameters should be
+set in a configuration block when you create the client.
+
+Old:
+```
+client = Grafeas.new credentials: my_credentials_object
+```
+
+New:
+```
+client = Grafeas.grafeas do |config|
+  config.credentials = my_credentials_object
+end
+```
+
+The configuration block is optional. If you do not provide it, or you do not
+set some configuration parameters, then the default configuration is used. See
+[Client Configuration](#client-configuration).
+
+### Provider Independence
+
+Older releases were tied to Google's Grafeas implementation, the Google Cloud
+Container Analysis service. Clients were configured to use Google's endpoint
+and credentials by default, though this could be overridden. You could also use
+Google-specific credential specifications such as the path to a Google service
+account keyfile.
+
+With the 1.0 release, the `grafeas` gem is provider-independent, and does not
+default to the Google implementation. You must specify an endpoint host, and
+you must provide gRPC-based credentials (such as a
+`GRPC::Core::ChannelCredentials` or a pre-constructed `GRPC::Core::Channel`
+object) when creating a client object. These properties can be set using the
+[configuration interface](#client-configuration) or by passing a block to the
+constructor as covered in [Creating Clients](#creating-clients).
+
+Old:
+```
+client = Grafeas.new credentials: "path/to/google/keyfile.json"
+```
+
+New:
+```
+client = Grafeas.grafeas do |config|
+  config.endpoint = "my-grafeas.example.com"
+  config.credentials = my_grpc_credentials
+end
+
+# or
+
+client = Grafeas.grafeas do |config|
+  config.credentials = my_existing_grpc_channel
+end
+```
+
+If you want to continue using Google's Cloud Container Analysis implementation,
+we recommend using the `google-cloud-container_analysis` gem, which retains the
+Google-specific defaults and features.
```

Comment:
This sentence does not explain that the Container Analysis client exposes a Grafeas client. Can you adapt the comment below ("You can obtain a Grafeas client from the Container Analysis connection") to appear here as well?